File: C/selectionsort/meta-llama-llama-4-maverick-17b-128e-instruct.c

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function to swap two elements
void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Function to perform Selection Sort
void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        int minIndex = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        if (minIndex != i) {
            swap(&arr[i], &arr[minIndex]);
        }
    }
}

// Function to generate a random array of size n
int* generateRandomArray(int n) {
    int* arr = (int*)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 1000;
    }
    return arr;
}

int main() {
    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }
    int arr[1000];
    int n = 0;

    while (fscanf(file, "%d", &arr[n]) == 1 && n < 1000) {
        n++;
    }

    selectionSort(arr, n);
    fclose(file);
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    return 0;
}

File: Java/selectionsort/meta_llama_llama_4_maverick_17b_128e_instruct.java
package Java.selectionsort;

public class meta_llama_llama_4_maverick_17b_128e_instruct {

    public static void selectionSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            // Swap the found minimum element with the first element of the unsorted part
            int temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
    }

    public static void main(String[] args) {
        // Generate 1000 random integers and store them in an array
        int[] arr = new int[1000];
        java.util.Random rand = new java.util.Random();
        for (int i = 0; i < 1000; i++) {
            arr[i] = rand.nextInt(10000);
        }

        // Perform selection sort
        long startTime = System.nanoTime();
        selectionSort(arr);
        long endTime = System.nanoTime();

        // You can verify the correctness of the sort by checking if the array is sorted
        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i] > arr[i + 1]) {
                throw new AssertionError("Array is not sorted");
            }
        }
    }

}
